src/client/entity-editor/common/entity-search-field-option.js

```diff
@@ -0,0 +1,129 @@
+ */
+
+import CustomInput from '../../input';
+import Entity from './entity';
+import LinkedEntity from './linked-entity';
+import PropTypes from 'prop-types';
+import React from 'react';
+import {Async as SelectAsync} from 'react-select';
+import ValidationLabel from '../common/validation-label';
+import _ from 'lodash';
+import makeImmutable from './make-immutable';
+import request from 'superagent-bluebird-promise';
+
+
+const ImmutableAsyncSelect = makeImmutable(SelectAsync);
+
+class EntitySearchFieldOption extends React.Component {
+	constructor(props) {
+		super(props);
+
+		// React does not autobind non-React class methods
+		this.fetchOptions = this.fetchOptions.bind(this);
+		this.isArea = this.isArea.bind(this);
+		this.entityToOption = this.entityToOption.bind(this);
+	}
+
+	/**
+	 * Determines whether an entity provided to the EntitySearch component is an
+	 * Area, using the present attributes.
+	 *
+	 * @param {Object} entity the entity to test
+	 * @returns {boolean} true if the entity looks like an Area
+	 */
+	isArea(entity) {
+		if (entity.type === 'Area') {
+			return true;
+		}
+
+		if (entity.gid) {
+			return true;
+		}
+
+		return false;
+	}
+
+	/**
+	 * Takes an entity and converts it to a format acceptable to react-select.
+	 *
+	 * @param {Object} entity the entity to convert
+	 * @returns {Object} the formatted data
+	 * @param {Array<LanguageOption>} languageOptions - The list of possible languages for an
+	 * entity.
+	 */
+	entityToOption(entity) {
+		const id = this.isArea(entity) ? entity.id : entity.bbid;
+		const languageId = _.get(entity, ['defaultAlias', 'languageId']);
+		const language = this.props.languageOptions.find(index => index.value === languageId);
+		return {
+			disambiguation: _.get(entity, ['disambiguation', 'comment']),
+			id,
+			language: language && language.label,
+			text: _.get(entity, ['defaultAlias', 'name']),
+			type: entity.type,
+			value: id
+		};
+	}
+
+	fetchOptions(query) {
+		return request
+			.get('/search/autocomplete')
+			.query({
+				collection: this.props.type,
+				q: query
+			})
+			.then((response) => ({
+				options: response.body.map(this.entityToOption)
+			}));
+	}
+
+	render() {
+		const labelElement = <ValidationLabel empty={this.props.empty} error={this.props.error}>{this.props.label}</ValidationLabel>;
+		return (
+			<CustomInput label={labelElement} tooltipText={this.props.tooltipText}>
+				<ImmutableAsyncSelect
+					labelKey="text"
+					loadOptions={this.fetchOptions}
+					optionComponent={LinkedEntity}
+					valueRenderer={Entity}
+					onBlurResetsInput={false}
+					{...this.props}
+				/>
+			</CustomInput>
+		);
+	}
+}
+
+EntitySearchFieldOption.displayName = 'EntitySearchFieldOption';
+EntitySearchFieldOption.propTypes = {
+	empty: PropTypes.bool,
+	error: PropTypes.bool,
+	label: PropTypes.string.isRequired,
+	languageOptions: PropTypes.array.isRequired,
+	tooltipText: PropTypes.string,
+	type: PropTypes.array.isRequired
```

Comment:
The `type` prop can either be a string or an array (for example in `client/entity-editor/relationship-editor/relationship-editor.js`), so the correct definition would be
```
type: PropTypes.oneOfType([
		PropTypes.string,
		PropTypes.arrayOf(PropTypes.string)
]).isRequired
```